components/k8s-model-server/README.md

```diff
@@ -162,3 +162,107 @@ You can learn more about [updating a Deployment](https://kubernetes.io/docs/conc
 [scaling a Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#scaling-a-deployment), and 
 [Pod Resources](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/) in the 
 Kubernetes documentation.
+
+
+
+### Use the served model
+
+The [inception-client](./inception-client) directory contains a Python script you can use to make a call against the deployed model.
+
+This script is intended to be run externally to the kubernetes cluster as a demonstration that the inception model is correctly being served.
+You can run the script either directly from a Python2 environment or in a Docker container.
+
+#### Setup
+
+You will require the external IP for the inception service as well as the port it is being hosted on. The inception service should be
+listed under the value you used for the `MODEL_NAME` parameter in the ksonnet component. You can find this information using
```

Comment:
What are we telling people to do in terms of accessing the service from outside the cluster? Are we telling them to expose the service publicly?

If we're exposing the service publicly lets make this very explicit and lets open up follow on issues to think about how we can point users in the direction of securing it using appropriate credentials.

I think we should try to avoid telling people to expose services publicly. So I lean towards suggesting solutions that rely on `kubectl proxy` or `kubectl port-forward`. In an advanced section we can point people in the direction of making this more accessible using IAP or other methods to make the service accessible from outside the cluster.

Another way to side step the issue of external access to your K8s cluster is to have people send the predictions from with the cluster using `kubectl run` or by running from a Jupyter notebook running inside the cluster.

Reply:
Currently, it looks like the tf serving component already exposes the service - when I run `kubectl get svc`, I see an external IP and I don't require any additional authentication against it.

For the purposes of the component guide, I do want to show users how to make calls to a kubeflow service from an external client. I agree with keeping things as they are and add a follow-on issue for authentication, as you suggested.

Reply:
I opened up #121 to change the default. I'd much prefer it if we didn't tell people to open up public IP's. I'd prefer it if our docs used kubectl and provided links or other instructions about how to add external IPs if desired.

Reply:
My idea was to merge this first so there's some client code in the readme, and then fix the service issue. But if you prefer to fix the issue before merging, that's fine too.

Added checklist on https://github.com/google/kubeflow/issues/121, will work on it in the next couple of days.

Reply:
I don't think you need to fix #121 even with type LoadBalancer you can access the gRPC from inside the cluster just like you would if it was a ClusterIp type. So we can go ahead and just make the instructions describe access without using an external IP. This way we don't have to go back and revise the instructions in the future.

Reply:
Good point, did as you said although getting images into kubernetes posed a bit of a dilemma - decided for a simple workflow in which they are provided with the image.